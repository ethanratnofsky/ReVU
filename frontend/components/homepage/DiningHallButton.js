import { Image, TouchableOpacity, Text, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useState, useEffect } from 'react';
import diningHallButtonStyles from './DiningHallButtonStyles';

import { ANONYMOUS_NAMES } from '../../constants';
import RATINGS from './demo';

const DiningHallButton = () => {
    // const navigation = useNavigation();

    const id = 0;
    const name = "Rand Dining Hall";
    const userId = "Wxdajkhdsa99";

    // TODO: Get ratings from backend
    //const ratings = RATINGS.filter(rating => rating.diningHallId === id);

    // const [ratings, setRatings] = useState([]);

    // const requestOptions = {
    //     method: 'GET',
    //     headers: {'Content-Type': 'application/json'}
    // };

    // useEffect(() => {
    //     fetch(`https://sleepy-reaches-22563.herokuapp.com/api/getAll/ratings/${id}`, requestOptions)
    //         .then(async response => {
    //             const isJson = response.headers.get('content-type')?.includes('application/json');
    //             const data = isJson && await response.json();

    //             if (!response.ok) {
    //                 const err = (data && data.message) || response.status;
    //                 return Promise.reject(error);
    //             }
    //             setRatings(data);
    //         }).catch(error => {
    //             console.log(error);
    //             alert("Could not get all ratings for overall rating.");
    //             console.log("Bad here in dining hall butotn");
    //         });
    // }, []);

    // TODO: Get ratings from backend
    const ratings = [
        {
            id: 10,
            userId: 2,
            diningHallId: 0,
            type: 'traffic',
            rating: 3,
        },
        {
            id: 11,
            userId: 2,
            diningHallId: 0,
            type: 'food',
            rating: 5,
        },
        {
            id: 12,
            userId: 2,
            diningHallId: 0,
            type: 'traffic',
            rating: 5,
        },
    ];

    let overallRating = (0.5 * (ratings.finFood + ratings.finTraffic));
    // let overallRating = null;
    if (ratings.numRatings === 0) {
        overallRating = null;
    }

    const handlePress = () => {
        navigation.navigate('Dining Hall', { id, name, userId });
    }

    return (
        <TouchableOpacity activeOpacity={0.6} style={diningHallButtonStyles.container} onPress={handlePress}>
            <Text style={diningHallButtonStyles.name}>{name}</Text>
            <View style={diningHallButtonStyles.ratingContainer}>
                <Text style={diningHallButtonStyles.rating}>{overallRating === null ? 'Unrated' : overallRating.toFixed(1)}</Text>
                {overallRating !== null && <Image style={diningHallButtonStyles.star} source={require('../../assets/images/gold-star.png')} />}
            </View>
        </TouchableOpacity>
    );
}

export default DiningHallButton;
