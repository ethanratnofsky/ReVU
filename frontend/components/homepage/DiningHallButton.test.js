import React from 'react';
const renderer = require('react-test-renderer');
import DiningHallButton from './DiningHallButton';

describe('<DiningHallButton />', () => {
  it('has 2 children', () => {
    const tree = renderer.create(<DiningHallButton />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<DiningHallButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });