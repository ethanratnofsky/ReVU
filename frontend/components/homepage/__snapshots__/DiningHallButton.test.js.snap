// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#CFAE70",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "space-between",
      "margin": 10,
      "opacity": 1,
      "padding": 15,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1C1C1C",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Rand Dining Hall
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1C1C1C",
          "fontFamily": "LibreCaslonText_400Regular",
          "fontSize": 16,
        }
      }
    >
      NaN
    </Text>
    <Image
      source={1}
      style={
        Object {
          "aspectRatio": 1,
          "bottom": 1,
          "left": 1,
          "position": "relative",
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
